### DISCUSSION TITLE
Unhandled Rejection (Range Error): Max call stack size exceeded

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2659] [General]
I get the following screen when I modify and save a tsx file in VSCode.  I believe it is trying to recompile and rerun the code.  Everything seems to still work OK, just reporting for information...

I am using 2.19.20

![max call stack](https://user-images.githubusercontent.com/83007711/140608088-d6a1616e-ba45-4ffc-8698-1fe33525f8d3.JPG)

The complete dump of message is in the attached file

[max call stack.txt](https://github.com/iTwin/itwinjs-core/files/7490352/max.call.stack.txt)

**Comment by kckst8** [comment]
Hi @ovaisa , can you give some more details about what you are attempting? Is this a clone of this repository that you are attempting to run?

**Reply by ovaisa** [reply]
I recently updated my iTwin version from 2.19.6 to 2.19.20.   I get this error when the app is running (while debugging in VSCode) and I make a change to the code and save the change.  Normally in the past, VSCode would restart the debugging session automatically after recompiling the code.  My app would restart from top.  Now I get this error and the app doesn't restart.  I just have to 'Stop Debugging Shift+F5' and 'Start Debugging F5' and everything seems to work fine.

This obviously only happens when I debugging and fixing/changing code, doesn't happen in production.  As I stated, it is not an issue for me as it is not stopping me from anything and the code seems to work fine once I restart debugging.  I just thought I  report it in case if has other ramifications.

**Reply by aruniverse** [reply]
Just to clarify, this is with an app you bootstrapped using the itwin web-viewer template?
Did you recently also update VSCode?

**Reply by ovaisa** [reply]
Yes, I bootstrapped a new app using the CRA template (2.19.20) and then migrated my code to this new app.  I did not update VSCode when I did this - I am using V1.62 (Oct 2021)

**Reply by aruniverse** [reply]
Side note, is there any reason you bootstrapped a new app using the cra template, and just copied the code over? Were you doing that to move up to the latest iTwinJs ver (2.19.20)? If so, you could directly update the dependencies in package.json.

Thanks for bringing this to our attention, we'll look into reproducing and resolving this issue.
